examples/querying_entity_traits.ipynb

```diff
@@ -0,0 +1,710 @@
+    "\n",
+    "This corresponds to an important philosophy of OpenAssetIO: when the user provides an entity reference, the host should not make assumptions about its provenance. It may be a reference to an existing entity, or a reference to a container, or a placeholder reference to an entity that doesn't exist yet, etc.\n",
+    "\n",
+    "As such, `entityTraits` should rarely appear in publishing code. However, there are use cases where it is valuable.\n"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "dbb66de3947d7b24"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Deciding whether an entity reference is appropriate\n",
+    "\n",
+    "Let's say we are in a widget used for publishing 3D models. The user provides an entity reference they want to publish to. \n",
+    "\n",
+    "Since we deal with 3D models, we expect this entity reference to support publishing a trait set of `{GeometryTrait.kId, LocatableContentTrait.kId}`. But the user gives us `logo_ref`..."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "8103bbcc9d83049c"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "entity_trait_set = manager.entityTraits(logo_ref, EntityTraitAccess.kWrite, context)\n",
+    "\n",
+    "helpers.display_result(entity_trait_set)\n",
+    "\n",
+    "# Check if the minimal trait set required for publishing is satisfied by the trait set we're planning to publish.\n",
+    "is_entity_supported = entity_trait_set <= {GeometryTrait.kId, LocatableContentTrait.kId}\n",
+    "\n",
+    "helpers.display_result(f\"Does our widget support this entity? {is_entity_supported}\")"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "c5cf5d230e7bc4d6"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "As expected, the trait set of the logo is not contained within our expected trait set, so our widget could flag an error to the user at this point.\n",
+    "\n",
+    "A host could skip this check and try to publish regardless, reporting any error (from `preflight` or `register`) to the user. The `preflight` method is intended, in part, as a validation step, so this workflow is not unreasonable. However, doing this would likely provide a worse UX, since the widget could not be pre-emptively disabled."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "5a0d4c198b48e907"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### Aside: the minimal trait set required for publishing \n",
+    "\n",
+    "In the previous example we see that the resulting trait set is smaller than that for `kRead`. This is because it is the minimal set of traits that _must_ be specified when publishing to this entity reference.\n",
```

Comment:
(And below) vaguely attempting to make it slightly more "qualities of the entity" to help reinforce that in this situation its about can this kind of thing be published to this ref vs being about the data.
```suggestion
    "In the previous example we see that the resulting trait set is smaller than that for `kRead`. This is because it is the minimal set of traits that the new entity _must_ possess when publishing to this entity reference.\n",
```

Reply:
Updated in e714b49e